dicthash/dicthash.py

```diff
@@ -71,10 +70,10 @@ def _generate_string_from_dict(d, blacklist, whitelist, prefix=''):
     if whitelist is None:
         whitelist = d.keys()
     if blacklist is not None:
-        whitelist = [key for key in whitelist if key not in blacklist]
+        whitelist = (key for key in whitelist if key not in blacklist)
+
     # Sort whitelist according to the keys converted to str
-    raw = [_unpack_value(d[key], prefix + str(key)) for key in sorted(whitelist, key=str)]
-    return ''.join(raw)
+    return ''.join(_unpack_value(d[key], prefix + str(key)) for key in sorted(whitelist, key=str))
```

Comment:
It is unrelated to the topic of the PR, but we should still change this:
prefix is a keyword argument to _unpack_value so we should use it like this. Thus, please insert a prefix=prefix+... here.